src/ucp/core/ucp_request.h

```diff
@@ -116,7 +115,21 @@ struct ucp_request {
                     uint8_t                   cbq_elem_on;
                     ucp_lane_map_t            lanes;     /* Which lanes need to be flushed */
                 } flush;
-
+                struct {
+                    uint64_t              remote_addr; /* Remote address */
+                    ucp_atomic_fetch_op_t op; /* Requested AMO */
+                    ucp_rkey_h            rkey;     /* Remote memory key */
+                    union {
+                        struct {
+                            uint32_t value;
+                            uint32_t *result;
+                        } amo32;
```

Comment:
is it possible to unite 32/64 bit fields in the request, also to reduce the amount of functions which handle 32/64 bit cases? for example, keep the value as 64-bit, and results as void*, the only difference would be in the progress function and not the request initialization?